File: tests/differential/fixtures/dataset_schema.ts
import { datasetSchema, schema } from "../../../src/schema";
import { Pipeline } from "../../../src/pipeline";

export function build(): Pipeline {
  const p = new Pipeline("Dataset Schema", { pipelineId: "dataset-schema" });
  p.sourceApi("Fetch", {
    url: "https://example.test/data",
    schema: datasetSchema(
      { id: schema.int64(), name: schema.string({ nullable: true }) },
      { additionalColumns: "closed" },
    ),
  });
  return p;
}

File: tests/differential/fixtures/dataset_schema.py
from brokoli import Pipeline, dataset_schema, source_api
from brokoli.ir import ir_digest, render_ir

with Pipeline("Dataset Schema", pipeline_id="dataset-schema") as p:
    source_api(
        "Fetch",
        url="https://example.test/data",
        schema=dataset_schema(
            {"id": {"kind": "int64"}, "name": {"kind": "string", "nullable": True}},
            additional_columns="closed",
        ),
    )

print(render_ir(p.to_json()), end="")
print("DIGEST " + ir_digest(p.to_json()))
